### PR TITLE
Feature/cli preview command

### PR DESCRIPTION
**Pull request by the38th**

### CODE REVIEW

File: lib/uffizzi/cli/preview.rb [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+      }
+
+      {
+        compose_file: compose_file_params,
+        dependencies: dependencies,
+      }
+    end
+
+    def deployment_name_valid?(deployment)
+      !(deployment =~ /^deployment-[0-9]+$/).nil?
```
Comment (moklidia) [inline lib/uffizzi/cli/preview.rb:262]: Very hard to read

File: lib/uffizzi/cli/preview.rb [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      return Cli::Common.show_manual(:create) if options[:help]
+
+      run(options, 'create', file_path, nil)
+    end
+
+    desc 'delete', 'delete'
+    def delete(deployment)
+      return Cli::Common.show_manual(:delete) if options[:help]
+
+      run(options, 'delete', nil, deployment)
```
Comment (moklidia) [inline lib/uffizzi/cli/preview.rb:37]: You are passing an argument (file_path as nil) that is not supposed to be in this method. There must be a better way, using ARGV or something.

File: lib/uffizzi/cli/preview.rb [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    end
+
+    desc 'describe', 'describe'
+    def describe(deployment)
+      return Cli::Common.show_manual(:describe) if options[:help]
```
Comment (moklidia) [inline lib/uffizzi/cli/preview.rb:42]: I think this can be moved to the upper level (`run`) as well

File: lib/uffizzi/cli/preview.rb [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        dependencies: dependencies,
+      }
+    end
+
+    def deployment_name_valid?(deployment)
+      return false unless deployment.start_with?('deployment-')
+      return false unless deployment.split('-').size == 2
+
+      deployment_id = deployment.split('-').last
+      deployment_id.to_i.to_s == deployment_id
```
Comment (moklidia) [inline lib/uffizzi/cli/preview.rb:265]: Why this check?

Comment (the38th) [reply lib/uffizzi/cli/preview.rb:265]: without this check, 'deployment-12s3' will count as valid

**Pull request merged**